src/lib.rs

```diff
@@ -107,16 +109,25 @@ impl<K, V> SequenceTrie<K, V>
     ///
     /// Returns `None` if the key did not already correspond to a value, otherwise the old value is
     /// returned.
-    pub fn insert<'key, I>(&mut self, key: I, value: V) -> Option<V>
-        where I: IntoIterator<Item = &'key K>
+    pub fn insert<'key, I, Q: 'key + ?Sized>(&mut self, key: I, value: V) -> Option<V>
```

Comment:
Note: need to bump the version, as this is a potentially breaking change. I left the version as-is for now so that the PR to iron/mount can reference this branch.